intro/scipy/index.rst

```diff
@@ -278,18 +278,70 @@ interpolation example.
 Optimization and fit: :mod:`scipy.optimize`
 -------------------------------------------
 
-Optimization is the problem of finding a numerical solution to a
-minimization or equality.
+:mod:`scipy.optimize` provides algorithms for root finding, curve fitting,
+and more general optimization.
 
-.. tip::
+Root Finding
+............
+
+:func:`scipy.optimize.root_scalar` attempts to find a root of a specified
+scalar-valued function (i.e., an argument at which the function value is zero).
+Like many :mod:`scipy.optimize` functions, the function needs an initial
+guess of the solution, which the algorithm will refine until it converges or
+recognizes failure.
+
+    >>> def f(x):
+    ...     return (x-1)*(x-2)
+    >>> res = sp.optimize.root_scalar(f, x0=0)  # x0 is the guess
+    >>> res
+      converged: True
+           flag: converged
+ function_calls: 12
+     iterations: 6
+           root: [ 1.000e+00]
+
+Note that only one the root at ``1.0`` is found. By inspection, we can tell
+that there is a second root at ``2.0``. We can direct the function toward a
+particular root by changing the guess or by passing a bracket that contains
```

Comment:
I would like to show that changing the guess to a point where the derivative is 0 can cause problems with the gradient-based methods. Would that be too much?